model_compression_toolkit/core/common/graph/memory_graph/max_cut_astar.py

```diff
@@ -0,0 +1,414 @@
+# Copyright 2022 Sony Semiconductor Israel, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import copy
+from typing import List, Tuple, Dict
+
+from model_compression_toolkit.core.common import BaseNode
+from model_compression_toolkit.core.common.constants import DUMMY_TENSOR, DUMMY_NODE
+from model_compression_toolkit.core.common.graph.memory_graph.cut import Cut
+from model_compression_toolkit.core.common.graph.memory_graph.memory_element import MemoryElements
+from model_compression_toolkit.core.common.graph.memory_graph.memory_graph import ActivationMemoryTensor, MemoryGraph
+
+
+class DummpyType:
+    """
+    A dummy type class to use for dummy nodes layer type.
+    """
+    pass
+
+
+class DummyBaseNodeGenerator:
+    """
+    A Dummy node which represents a "Side A" node (operation) in the memory graph for max cut AStar algorithm.
+    """
+    def __init__(self):
+        self.counter = 0
+
+    def __iter__(self) -> BaseNode:
+        """
+        A generator of sequentially named dummy nodes.
+        """
+        while True:
+            yield BaseNode(name=f"{DUMMY_NODE}_{self.counter}",
+                           framework_attr={},
+                           input_shape=tuple(),
+                           output_shape=tuple(),
+                           weights={},
+                           layer_class=DummpyType,
+                           has_activation=False)
+
+            self.counter += 1
```

Comment:
Maybe the dummy nodes should be placed somewhere else, more common? Because maybe someone will use these in the future in another alg.

Reply:
I'm not sure that I want to allow someone to use it in another context, this is very specific to this algorithm.